pkg/kubelet/volumemanager/cache/actual_state_of_world.go

```diff
@@ -525,7 +525,10 @@ func (asw *actualStateOfWorld) AddPodToVolume(markVolumeOpts operationexecutor.M
 	}
 
 	podObj, podExists := volumeObj.mountedPods[podName]
-	if !podExists {
+	if !podExists || podObj.volumeMountStateForPod == operationexecutor.VolumeMountUncertain {
+		// Add new mountedPod or update existing uncertain one - the new markVolumeOpts may
+		// have updated information. Especially reconstructed volumes (marked as uncertain
+		// during reconstruction) need update.
 		podObj = mountedPod{
```

Comment:
Does *not* doing this results in real breakage? (sorry just curious, I suspected it will but I don't know for sure).

Reply:
I don't know, you told me that we can't trust reconstructed volumes. And I think we should not trust Uncertain volumes either, so I update everything.